plutus-core/plutus-core/src/PlutusCore/TypeCheck.hs

```diff
@@ -51,41 +50,35 @@ builtinMeaningsToTypes ann =
 
 -- | Get the default type checking config.
 getDefTypeCheckConfig
-    :: (MonadError err m, AsTypeError err term uni fun ann, Typecheckable uni fun)
+    :: (MonadKindCheck err term uni fun ann m, Typecheckable uni fun)
     => ann -> m (TypeCheckConfig uni fun)
 getDefTypeCheckConfig ann = TypeCheckConfig <$> builtinMeaningsToTypes ann
 
 -- | Infer the kind of a type.
 inferKind
-    :: (MonadQuote m, MonadError err m, AsTypeError err term uni fun ann, ToKind uni)
+    :: MonadKindCheck err term uni fun ann m
```

Comment:
The main thing. `MonadKindCheck` doesn't contain `MonadQuote`. We don't need generation of unused variables for kind-checking and I think it's worth reflecting that in the types. Particularly since we can make constraints shorter and more meaningful in most cases with this strategy anyway.